### ISSUE TITLE
Rename 'username' to 'login'?

### ISSUE THREAD

**Issue by jaroel** [plone/plone.api#270]
In issue #204 @jensens proposed renaming 'username' to 'login' to differentiate more between the username/email and the userid.
I've done some work on https://github.com/plone/plone.api/compare/204_username_to_login already.

**Comment by mauritsvanrees** [comment]
+1, looks good at first glance. Agreed on renaming username to login.

What's with the email parameter? Where is that used? If this parameter is used for getting a user with a certain email address, this will only work when email is used as login. If that is not the case, then you cannot find a user by email address, so having an email parameter would be misleading. I would be in favor of renaming email to login then.

**Comment by jaroel** [comment]
The email parameter is in `plone.api.user.create`.

**Comment by mauritsvanrees** [comment]
Ah, right.

I would actually change username to userid there. Then in all cases the resulting user will have the userid that you have given. plone.api then just passes that to addMember. Only if email is used as login name, do we check for an email address.

Well... This is tricky. Both login and userid may be useful. And I guess that is true for all of the functions where we currently accept a username.

Say you have an add-on that creates a user like this:

```
user = api.user.create(email='jane@plone.org', login='jane')
```

In standard Plone you get userid jane, login jane.
When email is used as login, the login argument is completely ignored. Maybe accept this, but log a warning.

Note that the user id might need to be handled differently:
https://github.com/plone/plone.app.users/blob/2.3/plone/app/users/browser/register.py#L137
But that may be a separate issue.

**Comment by jaroel** [comment]
Yes, I ran into the same issue. I hadn't written it out, though.
I'm not convinced we should do this, now I see some more code.
